homeassistant/components/syncthing/config_flow.py

```diff
@@ -0,0 +1,81 @@
+"""Config flow for syncthing integration."""
+import logging
+
+import aiosyncthing
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_NAME, CONF_TOKEN, CONF_URL, CONF_VERIFY_SSL
+
+from .const import DEFAULT_NAME, DEFAULT_URL, DEFAULT_VERIFY_SSL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_URL, default=DEFAULT_URL): str,
+        vol.Required(CONF_TOKEN): str,
+        vol.Required(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_NAME] == data[CONF_NAME]:
```

Comment:
Can we use something else than name as id? Eg token or url? We don't want to use something arbitrary to guard against same entries.